### PR TITLE
improved imputation strategies

### PR DESCRIPTION
**Pull request by tvdboom**
#### Describe the changes you've made

Added new imputation strategies.

The new options for numerical columns are:
    - "drop": Drop rows containing missing values.
    - "mean": Impute with mean of column.
    - "median": Impute with median of column.
    - "mode": Impute with most frequent value.
    - "knn": Impute using a K-Nearest Neighbors approach.
    - int or float: Impute with provided numerical value.

The new options for categorical columns are:
    - "drop": Drop rows containing missing values.
    - "mode": Impute with most frequent value.
    - str: Impute with provided string.

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] Code style update (formatting, local variables)
- [x] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.
- [x] Any dependent changes have been merged and published in downstream modules.

### CODE REVIEW

File: pycaret/classification.py [hunk 1]
```diff
@@ -187,14 +187,21 @@ def setup(
         If None, no imputation of missing values is performed.
 
 
-    numeric_imputation: str, default = 'mean'
-        Missing values in numeric features are imputed with 'mean' value of the feature
-        in the training dataset. The other available option is 'median' or 'zero'.
+    numeric_imputation: int, float or str, default = 'mean'
```
Comment (ngupta23) [inline pycaret/classification.py:190]: @tvdboom Where does `iterative` imputation come into picture? How does this relate to `imputation_type`?

Comment (tvdboom) [reply pycaret/classification.py:190]: That didn't change wrt v2.x. There is a parameter in setup `imputation_type` that can be simple or iterative. If simple, you use the parameters `numeric_imputation` and `categorical_imputation`. If iterative, you use the parameters `iterative_imputation_iters`, `numeric_iterative_imputer` and `categorical_iterative_imputer`. I'll add to the docstring that they are ignored when imputation_type=iterative.

**Review by ngupta23 — APPROVED** [review]

File: pycaret/classification.py (hunk 1 above)
Comment (ngupta23) [reply pycaret/classification.py:190]: Looks good, but note that the same documentation will have to change in the Object-Oriented Interface setup method as well

Comment (vpadman1) [reply pycaret/classification.py:190]: Could you also look at SOM algorithm?

**Pull request merged**